music_consolidator/views/index.jade

```diff
@@ -3,19 +3,58 @@ extends layout
 block content
     h1= title
     a(href="/auth") Authenticate your user
-    div
+    div#spacer
+      div#loader-container
+        div#loader-box
+          div#loader 
+        div#loader-text Loading Title and Artist
+    div.clear
       form(action='/signup',method='post')
         div(data-role='fieldcontain')
           fieldset(data-role='controlgroup')
-            label(for='link') link
-               input(id='link',type='text',value='',placeholder='ex: https://www.youtube.com/mylink',name='link')
+            label(for='link') link: 
+            input(id='link',type='text',value='',placeholder='ex: https://www.youtube.com/mylink',name='link')
         div#artistdiv(data-role='fieldcontain')
           fieldset(data-role='controlgroup')
-            label(for='artist') artist
-               input(id='artist',type='text',value='',name='artist')
+            label(for='artist') artist: 
+            input(id='artist-input',type='text',value='',name='artist')
         div(id='title',data-role='fieldcontain')
           fieldset(data-role='controlgroup')
-            label(for='title') title
-               input(id='title',type='text',value='',name='title')
+            label(for='title') title: 
+            input(id='title-input',type='text',value='',name='title')
         div(data-role='fieldcontain')
           input(type='submit',value='Transfer Song',data-transition='fade', data-theme='c')
+
+    script.
+      var formLink = document.getElementById("link");
+      var val = "";
+      formLink.addEventListener('input', function(event) {
+        if (val != this.value) {
+          val = this.value
+          var match = val.search(/[0-9A-Za-z]{11}/g);
+          if (match != -1) {
+            getMetaData(val);
+          }
+        }
+      });
+      
+      function getMetaData(link) {
+        var loader = document.getElementById("loader-container");
+        loader.style.opacity = 1;
+        
+        var encodedLink = encodeURIComponent(link);
+        var request = new XMLHttpRequest();
+        
+        request.onreadystatechange = function(response) {
+          if (this.readyState == 4 && this.status == 200) {
+            var info = this.responseText.split(" - ");
```

Comment:
Is this the responseText of the request? What's `this.readyState == 4`?

Reply:
I think readyState is set to 4 when the request is completed and returned. status == 200 checks if the response was successful or not.